python/lsst/base/packages.py

```diff
@@ -285,25 +286,42 @@ def read(cls, filename):
         Parameters
         ----------
         filename : `str`
-            Filename from which to read.
+            Filename from which to read. The format is determined from the
+            file extension.  Currently support ``.pickle`` and ``.yaml``.
 
         Returns
         -------
         packages : `Packages`
         """
-        with open(filename, "rb") as ff:
-            return pickle.load(ff)
+        _, ext = os.path.splitext(filename)
+        if ext == ".pickle":
+            with open(filename, "rb") as ff:
+                return pickle.load(ff)
+        elif ext == ".yaml":
+            with open(filename, "r") as ff:
+                return yaml.load(ff, Loader=yaml.SafeLoader)
+        else:
+            raise ValueError(f"Unable to understand how to read file {filename}")
```

Comment:
raise ValueError(f"Unable to determine how to read file {filename} from extension {ext}")